runtime/src/main/kotlin/kotlin/native/concurrent/Worker.kt

```diff
@@ -39,14 +39,11 @@ public inline class Worker @PublishedApi internal constructor(val id: Int) {
         public fun start(errorReporting: Boolean = true): Worker = Worker(startInternal(errorReporting))
 
         /**
-         * Return the current worker, if known, null otherwise. null value will be returned in the main thread
-         * or platform thread without an associated worker, non-null - if called inside worker started with
-         * [Worker.start].
+         * Return the current worker. Worker context is accessible to any valid Kotlin context,
+         * but only actual active worker produced with [Worker.start] automatically processes execution requests.
+         * For other situations [processQueue] must be called explicitly to process request queue.
          */
-        public val current: Worker? get() {
-            val id = currentInternal()
-            return if (id != 0) Worker(id) else null
-        }
+        public val current: Worker get() = Worker(currentInternal())
```

Comment:
Btw, `currentInternal()` can still return `0`, for example in deallocation hooks during final GC.

Reply:
Also when `WITH_WORKERS` is false.

Reply:
No for second comment, 
```
KInt currentWorker() {
  ThrowWorkerUnsupported();
  return 0;
}``` will throw an exception in  first case as well.

Reply:
> No for second comment,

Ok, I see.

> will throw an exception in first case as well.

Could you clarify?

Reply:
I mean, I fill fix the code, what I just did in 6990ad0c9